ls1intum/Artemis
# Performance improvement for courses on dashboard

Due to a change in one hibernate query, the performance of the courses for dashboard REST call (to initially load all data for the course overview) has deteriorated. 
The problem was the use of `or` within a join condition, which could not be optimized by the database.

The performance of the initial courses for dashboard REST call was 5x - 10x slower than usual.

This PR fixes the problem by splitting the problematic query into two distinct ones, one for individual and one for team exercises. It also avoids two eager attributes in Exercise which can be fetched differently.

### Checklist

### Steps for Testing
1. Open developer tools
2. Log in to Artemis
3. Navigate between Course Overview and Course Administration
4. Check that the REST call courses for dashboard is less than 100ms.
5. Check that categories and team config are displayed correctly in course overview (students) and when editing exercises (course administration)

No UI changes, no changes to tests needed
